packages/geo/src/lib/layer/layer-legend/layer-legend.component.ts

```diff
@@ -60,4 +61,21 @@ export class LayerLegendComponent implements OnInit, OnDestroy {
     this.legendItems$.next(legendItems);
   }
 
+  computeItemTitle(layerLegend): Observable<string> {
+    const layerOptions = this.layer.dataSource.options as any;
+    if (layerOptions.type !== 'wms') {
+      return of(layerLegend.title);
+    /*} else if (!layerOptions.optionsFromCapabilities) {
+      return of(layerLegend.title);*/
+    } else {
+      const layers = layerOptions.params.layers.split(',');
+      const localLayerOptions = JSON.parse(JSON.stringify(layerOptions)); // to avoid to alter the original options.
+      localLayerOptions.params.layers = layers.find(layer => layer === layerLegend.title);
+      return this.capabilitiesService
+        .getWMSOptions(localLayerOptions)
+        .pipe(map(wmsDataSourceOptions => {
+          return (wmsDataSourceOptions as any)._layerOptionsFromCapabilities.title;
```

Comment:
Is casting as any absolutely needed?

Reply:
as in layer.service.ts, line 62.

Reply:
if not :

Property '_layerOptionsFromCapabilities' does not exist on type 'WMSDataSourceOptions'.ts

Reply:
Should _layerOptionsFromCapabilities exist on WMSDataSourceOptions then?

Reply:
Added to interface.